### PR TITLE
add exclude filtering

### LINKED ISSUE #1: Support filter by Module Id
3rd Year Students have timetables which show modules taken in other semesters. It would be good to filter these by selecting the relevant Module IDs.

### PR DESCRIPTION
**Pull request by MathematicalDessert**
This resolves #1.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "python.pythonPath": "/usr/bin/python"
+    "python.pythonPath": "C:\\Python39\\python.exe"
```
Comment (kcxt) [inline .vscode/settings.json]: lol nice, probably should drop .vscode from git.

File: main.py [hunk 2]
```diff
@@ -24,18 +24,26 @@ def eventIsDifferentSubHouse(name, g, p):
 def index(stu_hash):
     print(request.url)
     g = request.args.get('group')
+    excludeList = request.args.get('exclude')
```
Comment (kcxt) [inline main.py:27]: Can you update the readme to reflect the API changes?

File: main.py [hunk 3]
```diff
@@ -24,18 +24,26 @@ def eventIsDifferentSubHouse(name, g, p):
...
     hide_optional = request.args.get('hide_optional')
     hide_optional = hide_optional if not hide_optional == None else "f"
     
-    if g == None or not len(g.split(":")) == 2:
-        return "Your query string wasn't in the correct format"
-    p = g.split(":")[1].lower()
-    g = g.split(":")[0].lower()
-    if not g in groups.keys() or not p in ['a', 'b']:
-        return "Your query string wasn't in the correct format"
+    # g is now optional
```
Comment (kcxt) [inline main.py:31]: Do we need G anymore? Can just drop it entirely.

File: main.py [hunk 4]
```diff
@@ -45,17 +53,23 @@ def index(stu_hash):
...
-        del removeGroups[removeGroups.index(groups[g])]
-        pattern = re.compile("^((?!{}|{}).)+$".format(removeGroups[0], removeGroups[1]))
-        if pattern.match(name): #This means the event name contains either the name of this users house, or no houses
-            if re.compile("^.*{}.*$".format(groups[g])).match(name):
-                if eventIsDifferentSubHouse(name, g, p):
+
+        if name[0:6] in (str.lower() for str in toExclude):
+            toRemove.append(ev)
+
+        if g != None:
```
Comment (kcxt) [inline main.py:60]: Dropping g lets us drop this stuff to.

**Review by kcxt — CHANGES_REQUESTED** [review]
pog PR thanks PK

File: .vscode/settings.json (hunk 1 above)
Comment (MathematicalDessert) [reply .vscode/settings.json]: Yeah, I was going to add it to the .gitignore...I guess I will

File: main.py (hunk 3 above)
Comment (MathematicalDessert) [reply main.py:31]: We could allow for first and second year students to use the API too, and by the looks of it the groupings still exist. Lmk what you think

image

File: main.py (hunk 4 above)
Comment (MathematicalDessert) [reply main.py:60]: See above.

File: main.py (hunk 3 above)
Comment (kcxt) [reply main.py:31]: Sure

**Comment by kcxt** [comment]
Sweet!

**Pull request merged**